### PR TITLE
osd: ReplicatedBackend::prep_push() remove redundant variable assignments

### PR DESCRIPTION
**Pull request by TsaiJin**
Signed-off-by: Jin Cai <caijin.caij@alibaba-inc.com>

### CODE REVIEW

**Comment by TsaiJin** [comment]
@tchaikov Please help me take a look.

File: src/osd/ReplicatedBackend.cc [hunk 1]
```diff
@@ -1572,8 +1572,8 @@ void ReplicatedBackend::prep_push(
   pi.recovery_info.version = version;
   pi.recovery_progress.first = true;
   pi.recovery_progress.data_recovered_to = 0;
-  pi.recovery_progress.data_complete = 0;
-  pi.recovery_progress.omap_complete = 0;
+  pi.recovery_progress.data_complete = false;
```
Comment (tchaikov) [inline src/osd/ReplicatedBackend.cc]: @TsaiJin i think we can just remove following lines. as they are already initialized by the default ctor of `ObjectRecoveryProgress`
```c++
  pi.recovery_progress.first = true;
  pi.recovery_progress.data_recovered_to = 0;
  pi.recovery_progress.data_complete = 0;
  pi.recovery_progress.omap_complete = 0;
```

**Comment by tchaikov** [comment]
could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes, in this case, it would be "osd: ".

File: src/osd/ReplicatedBackend.cc (hunk 1 above)
Comment (TsaiJin) [inline src/osd/ReplicatedBackend.cc]: Yes, you are right. I will modify it by your advice.

**Comment by TsaiJin** [comment]
@tchaikov I remove these unnecessary variable assignments by your advice.

**Comment by liupan1111** [comment]
retest this please

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2017-04-27_22:07:22-rados-wip-yuri-testing_2017_4_28_2-distro-basic-smithi/

http://pulpito.ceph.com/yuriw-2017-04-27_22:07:22-rados-wip-yuri-testing_2017_4_28_2-distro-basic-smithi/

**Pull request merged**